src/backend/api/Fusion.Resources.Logic/Requests/Commands/ResourceAllocationRequest/Allocation/ProvisionAllocationRequest.cs

```diff
@@ -46,6 +46,8 @@ protected override async Task Handle(ProvisionAllocationRequest request, Cancell
                         if (dbRequest.OrgPositionId is null)
                             throw new InvalidOperationException("Position id cannot be empty when provisioning request");
 
+                        await DeleteResourceOwnerNotes(request.RequestId, cancellationToken);
```

Comment:
Would not a better place be to use the base Provision command? Here we are missing the resource owner requests and any other that might have comments.

Could add a `INotification` in the provision command, 

```
if (provisionSuccessfully)  {
  await mediator.Publish(new RequestProvisioned(requestId));
} else {
  await mediator.Publish(new RequestProvisioningFailed(requestId);
}

public DeleteNotesHandler : INotificationHandler<RequestProvisioned> 
{
  public async Task Handle(notification) {
     var comments = await GetCommentsAsync();
     db.RemoveRange(comments);
     db.SaveChanges();
  }
} 
```

The handler could just be in the logic library. But this event will be usefull eventually when we want to publish subscription events and send notifications...